ledger-api/grpc-definitions/com/daml/ledger/api/v1/transfer_command.proto

```diff
@@ -0,0 +1,192 @@
+  reserved 7; // was maximum_record_time
+
+  oneof command {
+    TransferOutCommand transfer_out_command = ;
+    TransferInCommand transfer_in_command = ;
+    TransferOutInCommand transfer_out_in_command = ;
+  }
+
+  // this is replaced by the transfer commands above
+  // Individual elements of this atomic command. Must be non-empty.
+  // Required
+  //  repeated Command commands = 8;
+
+  // Specifies the deduplication period for the change ID.
+  // If omitted, the participant will assume the configured maximum deduplication time (see
+  // ``ledger_configuration_service.proto``).
+  oneof deduplication_period {
+    // Specifies the length of the deduplication period.
+    // Same semantics apply as for `deduplication_duration`.
+    // Must be non-negative. Must not exceed the maximum deduplication time (see
+    // ``ledger_configuration_service.proto``).
+    google.protobuf.Duration deduplication_time = [deprecated = true];
+
+    // Specifies the length of the deduplication period.
+    // It is interpreted relative to the local clock at some point during the submission's processing.
+    // Must be non-negative. Must not exceed the maximum deduplication time (see
+    // ``ledger_configuration_service.proto``).
+    google.protobuf.Duration deduplication_duration = ;
+
+    // Specifies the start of the deduplication period by a completion stream offset (exclusive).
+    // Must be a valid LedgerString (as described in ``ledger_offset.proto``).
+    string deduplication_offset = ;
+  }
+
+  // TODO are we sure we do not need ledger_time bounds for transfers?
+  // Lower bound for the ledger time assigned to the resulting transaction.
+  // Note: The ledger time of a transaction is assigned as part of command interpretation.
+  // Use this property if you expect that command interpretation will take a considerate amount of time, such that by
+  // the time the resulting transaction is sequenced, its assigned ledger time is not valid anymore.
+  // Must not be set at the same time as min_ledger_time_rel.
+  // Optional
+  // google.protobuf.Timestamp min_ledger_time_abs = 10;
+
+  // Same as min_ledger_time_abs, but specified as a duration, starting from the time the command is received by the server.
+  // Must not be set at the same time as min_ledger_time_abs.
+  // Optional
+  // google.protobuf.Duration min_ledger_time_rel = 11;
+
+  // Set of parties on whose behalf the command should be executed.
+  // If ledger API authorization is enabled, then the authorization metadata must authorize the sender of the request
+  // to act on behalf of each of the given parties.
+  // This field supersedes the ``party`` field. The effective set of parties on whose behalf the command
+  // should be executed is the union of all parties listed in ``party`` and ``act_as``, which must be non-empty.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional
+  repeated string act_as = ;
```

Comment:
Will multi-party actAs transfers work?

Reply:
Why would I do that? even if we require that you are a signatory to transfer I don’t understand why I would need to specify multiple parties

Reply:
User management does not automatically pick the party. It still requires that you specify it explicitly and then the auth service checks that the party you specify is a subset of the parties you have claims for.

The JSON API does support automatically picking the party but you can still specify it manually. So I'd just error out if there is more than one actAs party there and require users to override it.

Reply:
I am not sure how multi-party act_as would work. If I recall correctly @rgugliel-da mentioned something about choosing an appropriate user if a list is provided :thinking:

Reply:
As discussed this will be singular, and naming changes to submitter